src/dawn/engine_dawn.cc

```diff
@@ -932,8 +933,29 @@ Result EngineDawn::DoPatchParameterVertices(
   return Result("Dawn:DoPatch not implemented");
 }
 
-Result EngineDawn::DoBuffer(const BufferCommand*) {
-  return Result("Dawn:DoBuffer not implemented");
+Result EngineDawn::DoBuffer(const BufferCommand* command) {
+  Result result;
+
+  // TODO(SarahM0): Make this work for compute pipeline
+  RenderPipelineInfo* render_pipeline = GetRenderPipeline(command);
+  if (!render_pipeline)
+    return Result("DoBuffer invoked on invalid or missing render pipeline");
+  if (!command->IsSSBO() && !command->IsUniform())
+    return Result("DoBuffer not supported buffer type");
+
+  ::dawn::Buffer* dawn_buffer = buffer_map_[command->GetBinding()];
+  if (dawn_buffer) {
+    Buffer* amber_buffer = command->GetBuffer();
+    if (amber_buffer) {
+      amber_buffer->SetDataWithOffset(command->GetValues(),
+                                      command->GetOffset());
+
+      dawn_buffer->SetSubData(command->GetOffset(),
+                              amber_buffer->GetSizeInBytes(),
+                              amber_buffer->ValuePtr()->data());
```

Comment:
I am not versed on `dawn::Buffer::SetSubData(offset, size_in_byte, data_ptr)`, but does it mean

"Copy data_ptr[0 : size_in_byte] to dawn::Buffer::buffer[offset : offset + size_in_byte]"?

Reply:
Yes, that's right.